### ISSUE TITLE
[core] Slow execution at SynthUtils.toCloudFormation on v1.57.0

### ISSUE THREAD

**Issue by idm-ryou** [aws/aws-cdk#10213]
Our unit test execution takes considerably long time, when we upgrade CDK from v1.56.0 to v1.57.0. 

- v1.56.0: `$ jest` took **59 seconds**
- v1.57.0: `$ jest` took **673 seconds**

I suspect that changes from https://github.com/aws/aws-cdk/pull/9342 affected our codes.
Is there any workaround to make it faster like previous version?

### Reproduction Steps

Our test code something looks like followings:
```ts
import * as cdk from "@aws-cdk/core";
import "@aws-cdk/assert/jest";
import { SynthUtils } from "@aws-cdk/assert";
import { app } from "./app";

describe("app", () => {
  const stacks = app.node.children
    .filter(cdk.Stack.isStack);

  test("stack templates match snapshots", () => {
    stacks.forEach((stack) => {
      expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot(
        stack.stackName
      );
    });
  });
});
```

The `app` is pretty large and it has 55 stacks and the snapshot file has about 23,000 lines.

### Environment

  - **CLI Version      :1.57.0**
  - **Framework Version:1.57.0**
  - **Node.js Version:v12.18.3** 
  - **OS               :Linux(Docker)**
  - **Language (Version):TypeScript (3.8.3)** 

### Other

--- 

This is :bug: Bug Report

**Comment by NetaNir** [comment]
Hi! 

Sorry about that. While we investigate can you modify

**Comment by rix0rrr** [comment]
Investigating:

* A lot of time is spent in `collectRuntimeInformation` (specifically: `findNpmPackage`).
* Besides that, `captureStackTrace` in `PostResolveToken` is quite expensive (and we create A LOT of those).

The first finding will be addressed in https://github.com/aws/aws-cdk/pull/10306.

The second one is probably an easy enough fix. I thought we had disabled stack trace collection for tests using an environment variable. Maybe that change got reverted by accident?
